### DISCUSSION TITLE
How does the app access its own generated files？

### DISCUSSION THREAD

**Discussion by ccinsz** [beeware/briefcase#1357] [Q&A]
When the app access its own generated files, it raises Permission denied. How does the app access its own generated files？

**Comment by freakboy3742** [comment]
It's very difficult to answer questions when we don't have any context for those questions.

An app shouldn't have any difficulty accessing files, provided they're written in a location where the operating system has read access. Generally, the locations provided by `toga.App.paths` will be reliable readable base paths. They *may* also be writable, depending on the platform

Beyond that, it's difficult to answer why you're getting a Permission Denied error without knowing what you're trying to do, and on what platform you're doing it.

Based on past queries, I'm *assuming* this is on Android, and I'm *assuming* you're trying to write to an arbitrary file system location (possibly external storage). This may not be possible; Android requires the use of specific APIs to access files in some locations (such as external storage), and requires that the app ask for additional permissions before using those APIs. However, without details, it's impossible to guide you further.

**Reply by ccinsz** [reply]
Yes, the android app generates a file, and want to write in this file, but it raises  Read-only file system. I want to know how the android app can read and write its own file?

**Reply by mhsmith** [reply]
You can read and write to the directory given by [`paths.data`](https://toga.readthedocs.io/en/latest/reference/api/resources/app_paths.html). This is writable on all platforms.

**Reply by ccinsz** [reply]
Does this directory also have execute permission?

**Reply by mhsmith** [reply]
Yes, but that doesn't mean you can call an executable (#1359).
